posthog/storage/test/test_object_storage.py

```diff
@@ -157,3 +159,54 @@ def test_can_safely_copy_objects_from_unknown_prefix(self) -> None:
                 "test_storage_bucket/a_shared_prefix/b",
                 "test_storage_bucket/a_shared_prefix/c",
             ]
+
+    def test_read_bytes_with_byte_range(self):
+        # Setup
+        mock_client = MagicMock()
+        mock_body = MagicMock()
+
+        # For the first test, return a specific content
+        mock_body.read.return_value = b"test content"
+        mock_client.get_object.return_value = {"Body": mock_body}
+        storage = ObjectStorage(mock_client)
+
+        # Test with both first_byte and last_byte
+        result = storage.read_bytes("test-bucket", "test-key", first_byte=5, last_byte=10)
+
+        # Assert
+        mock_client.get_object.assert_called_with(Bucket="test-bucket", Key="test-key", Range="bytes=5-10")
+        self.assertEqual(result, b"test content")
+
+        # Test with only first_byte
+        result = storage.read_bytes("test-bucket", "test-key", first_byte=5)
+
+        # Assert
+        mock_client.get_object.assert_called_with(Bucket="test-bucket", Key="test-key", Range="bytes=5-")
+        self.assertEqual(result, b"test content")
```

Comment:
not at all important but the `assert result == b"test content"` is more pythonic and actually gives better error messages just fyi